### PR TITLE
implement data retention for boards

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Implement Data Retention for Boards
- mattermost-webapp and mattermost-server already have necessary changes.

#### Ticket Link

  Fixes https://community.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/b4ctmm917q3rzdfi451qtuxjzcw/vezy9zbpm1tnmfecbuoopjgsgqc/c88h9o9twcbgw787cd61r5a3wkw

### CODE REVIEW

File: mattermost-plugin/Makefile [hunk 1]
```diff
@@ -215,6 +215,7 @@ detach: setup-attach
 ## Runs any lints and unit tests defined for the server and webapp, if they exist.
 .PHONY: test
 test: webapp/node_modules
+export FB_UNIT_TESTING=1
```
Comment (Pinjasaur) [inline mattermost-plugin/Makefile:218]: do we always want this?

Comment (sbishel) [reply mattermost-plugin/Makefile]: Updated...

File: server/services/config/config.go [hunk 2]
```diff
@@ -50,6 +50,8 @@ type Configuration struct {
 	LocalModeSocketLocation  string            `json:"localModeSocketLocation" mapstructure:"localModeSocketLocation"`
 	EnablePublicSharedBoards bool              `json:"enablePublicSharedBoards" mapstructure:"enablePublicSharedBoards"`
 	FeatureFlags             map[string]string `json:"featureFlags" mapstructure:"featureFlags"`
+	EnableDataRetention      bool              `json:"enable_data_retention" mapstructure:"enable_data_retention"`
```
Comment (wiggin77) [inline server/services/config/config.go:53]: Should this be captured for metrics?

Comment (sbishel) [reply server/services/config/config.go:53]: This is actually captured in the mattermost-server, as this is set in that conf file. However, I am going to create a card for pass config setting in daily metrics, we currently don't do that.

**Review by wiggin77 — APPROVED** [review]
LGTM 👍

**Review by Pinjasaur — APPROVED** [review]
LGTM

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Pull request merged**